drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/referral_postings_loader.rb

```diff
@@ -0,0 +1,269 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::AcHmis::Importers::Loaders
+  class ReferralPostingsLoader < BaseLoader
+    def perform
+      # warning- this destroys all referrals regardless of data source
+      referral_class.destroy_all if clobber
+
+      import_enrollment_records
+      import_referral_records
+      import_referral_posting_records
+      import_referral_household_members_records
+      assign_unit_occupancies
+    end
+
+    POSTINGS_FILENAME = 'ReferralPostings.csv'.freeze
+    HOUSEHOLD_MEMBERS_FILENAME = 'ReferralHouseholdMembers.csv'.freeze
+
+    def runnable?
+      super &&
+      reader.file_present?(POSTINGS_FILENAME) &&
+      reader.file_present?(HOUSEHOLD_MEMBERS_FILENAME)
+    end
+
+    protected
+
+    def referral_class
+      HmisExternalApis::AcHmis::Referral
+    end
+
+    def supports_upsert?
+      true
+    end
+
+    #
+    # Data Access
+    #
+    def posting_rows
+      reader.rows(POSTINGS_FILENAME)
+    end
+
+    def household_member_rows
+      reader.rows(HOUSEHOLD_MEMBERS_FILENAME)
+    end
+
+    #
+    # Importers
+    #
+    def import_enrollment_records
+      # and easily do bulk upsert here due to dependent WIP records
+      without_paper_trail do
+        build_enrollment_rows.each(&:save!)
+      end
+    end
+
+    def import_referral_records
+      ar_import(
+        referral_class,
+        build_referral_records,
+        on_duplicate_key_update: { conflict_target: :identifier, columns: :all },
+      )
+    end
+
+    def import_referral_posting_records
+      ar_import(
+        HmisExternalApis::AcHmis::ReferralPosting,
+        build_posting_records,
+        on_duplicate_key_update: { conflict_target: :identifier, columns: :all },
+      )
+    end
+
+    def import_referral_household_members_records
+      ar_import(
+        HmisExternalApis::AcHmis::ReferralHouseholdMember,
+        build_household_member_records,
+        on_duplicate_key_update: { conflict_target: [:client_id, :referral_id], columns: :all },
+      )
+    end
+
+    #
+    # Builders
+    #
+
+    # create wip enrollments for accepted pending postings
+    def build_enrollment_rows
+      accepted_pending_status = HmisExternalApis::AcHmis::ReferralPosting.statuses.fetch('accepted_pending_status')
+      enrollment_coc_by_project_id = Hmis::Hud::ProjectCoc
+        .where(data_source: data_source)
+        .pluck(:project_id, :project_coc_id)
```

Comment:
```suggestion
        .pluck(:project_id, :coc_code)
```

Reply:
EnrollmentCoC 3.16.1 is the coc code itself https://files.hudexchange.info/resources/documents/HMIS-CSV-Format-Specifications-2024.pdf

Reply:
oh oops, thanks!